lib/cc/config/prepare.rb

```diff
@@ -0,0 +1,68 @@
+module CC
+  module Config
+    class Prepare
+      class NoPrepareNeeded
+        def fetch
+          []
+        end
+      end
+
+      attr_reader :fetch
+
+      def self.from_yaml(data)
+        if data.present?
+          fetch = Fetch.from_yaml(data.fetch("fetch", []))
+
+          new(fetch: fetch)
+        else
+          NoPrepareNeeded.new
+        end
+      end
+
+      def initialize(fetch:)
+        @fetch = fetch
+      end
+
+      class Fetch
+        def self.from_yaml(data)
```

Comment:
Based on the mess we got ourselves into with `cc-yaml`, I'm inclined to say the "parse" stage should effectively produce a plain YAML object (maybe with some very light type-checking/coercing), and extracting semantic meaning from that YAML is a separate stage. Which it looks to me like what you have here, so I think this is a good direction.